src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs

```diff
@@ -39,9 +47,7 @@ public sealed partial class NamedPipeServerStream : PipeStream
                 throw new ArgumentOutOfRangeException(nameof(pipeName), SR.ArgumentOutOfRange_AnonymousReserved);
             }
 
-            PipeSecurity pipeSecurity = null;
-
-            if (IsCurrentUserOnly)
+            if (IsCurrentUserOnly && pipeSecurity == null)
```

Comment:
In CoreFx an assert will be good to ensure that whenever `IsCurrentUserOnly` is true `pipeSecurity` is also null. In Mono what will be the behavior for the constructors that the respective parameters?

Reply:
The legacy constructors won't allow `CurrentUserOnly` mode if `pipeSecurity != null`.

Reply:
Could you please add a Debug.Assert - other than that LGTM.

Reply:
Added the assert here: https://github.com/dotnet/corefx/pull/31118/files#diff-a7e95545fe2b63c35aaee6c4c3de3cffR53

Reply:
Ops, missed that.